cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -933,6 +933,38 @@ TEST(ExecPlanExecution, SourceGroupedSum) {
   }
 }
 
+TEST(ExecPlanExecution, SourceMinMaxScalar) {
+  for (bool parallel : { false, true }) {
+    SCOPED_TRACE(parallel ? "parallel/merged" : "serial");
+
+    auto input = MakeGroupableBatches(/*multiplicity=*/parallel ? 100 : 1);
+    auto minmax_opts = std::make_shared<ScalarAggregateOptions>();
+    auto expected_value = StructScalar::Make(
```

Comment:
nit, but doesn't ScalarFromJSON handle StructScalar directly?

Reply:
https://github.com/apache/arrow/blob/8042f001fbadd2d4bb395e42a94c76fe7ee01d3b/cpp/src/arrow/ipc/json_simple.cc#L659-L661

should be `{"min": -8, "max": 12}`